test/ConnectedRouter.test.js

```diff
@@ -123,6 +123,19 @@ describe('ConnectedRouter', () => {
       expect(onLocationChangedSpy.mock.calls).toHaveLength(3)
     })
 
+    it('supports location state and key', () => {
+      mount(
+        <Provider store={store}>
+          <ConnectedRouter {...props}>
+            <Route path="/" render={() => <div>Home</div>} />
+          </ConnectedRouter>
+        </Provider>
+      )
+      props.history.push({ pathname: '/new-location', state: { foo: 'bar' } })
```

Comment:
Can you test `key` as well?

Reply:
I removed `location.key` from the perimeter of this PR, as it seems to be a private API of the `history` package.